### PR TITLE
PRIS-144: Support XLSX Excel File

### PR DESCRIPTION
**Pull request by rssntn67**
I have updated the XLS plugin to use org.apache.poi.

### CODE REVIEW

**Comment by rssntn67** [comment]
https://issues.opennms.org/browse/PRIS-144

**Comment by rssntn67** [comment]
Why build fails...Why build docs fails? Ronny please help me!

**Comment by indigo423** [comment]
@rssntn67 was able to fix the docs build issue. Asciidoctor is running in --safe mode which causes an false/positive error. When I pushed the change to your branch the docs pass now but the smoke test with the sample XLS source fails to throw an NPE on the Jetty:

```
07:49:33.031 [qtp1622006612-14] WARN org.opennms.pris.driver.RequisitionProviderHandler - Request failed
java.lang.NullPointerException: null
	at org.opennms.opennms.pris.plugins.xls.source.XlsSource.dump(XlsSource.java:183)
	at org.opennms.pris.RequisitionGenerator.generate(RequisitionGenerator.java:152)
	at org.opennms.pris.driver.RequisitionProviderHandler.handle(RequisitionProviderHandler.java:68)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:170)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:219)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:530)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:347)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:256)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:247)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:140)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:382)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:708)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:626)
	at java.lang.Thread.run(Thread.java:748)
07:49:33.344 [qtp1622006612-17] WARN org.opennms.pris.driver.RequisitionProviderHandler - Request failed
java.lang.NullPointerException: null
	at org.opennms.opennms.pris.plugins.xls.source.XlsSource.dump(XlsSource.java:183)
	at org.opennms.pris.RequisitionGenerator.generate(RequisitionGenerator.java:152)
	at org.opennms.pris.driver.RequisitionProviderHandler.handle(RequisitionProviderHandler.java:68)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:170)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:219)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:530)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:347)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:256)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:247)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:140)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:382)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:708)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:626)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by indigo423** [comment]
@fooker can you please help to review the code?

**Comment by indigo423** [comment]
@mvrueden seems like @fooker is not available, can you help out with a code review?

File: opennms-pris-plugins/opennms-pris-plugins-xls/pom.xml [hunk 1]
```diff
@@ -21,12 +21,16 @@
...
-        <dependency>
-            <groupId>net.sourceforge.jexcelapi</groupId>
-            <artifactId>jxl</artifactId>
-            <version>${jexcelApiVersion}</version>
-        </dependency>
-
+    <dependency>
+        <groupId>org.apache.poi</groupId>
+        <artifactId>poi</artifactId>
+        <version>3.17</version>
```
Comment (mvrueden) [inline opennms-pris-plugins/opennms-pris-plugins-xls/pom.xml]: Can we define a "apachePoiVersion" property and use it here instead? This seems to be the current behaviour, anyways.

Comment (mvrueden) [reply opennms-pris-plugins/opennms-pris-plugins-xls/pom.xml]: Besides that, the `JexcelApiVersion` from line 19 is not used anymore and can probably be removed in the code as well.

**Comment by mvrueden** [comment]
Besides the fact, that IDE-specific files are modified and part of the code, this looks fine to me. However, bamboo seems unhappy.

File: opennms-pris-plugins/opennms-pris-plugins-xls/pom.xml (hunk 1 above)
Comment (rssntn67) [reply opennms-pris-plugins/opennms-pris-plugins-xls/pom.xml]: yup thanks

**Comment by rssntn67** [comment]
Still problem building..but nevertheless on my mac compiles.....

**Comment by mfuhrmann** [comment]
@rssntn67 Seems not to work anymore: https://opennms.discourse.group/t/xlsx-as-source-in-pris-server/888

**Pull request merged**